crux-mir/test/conc_eval/vec/of_u8_array.rs

```diff
@@ -1,19 +1,28 @@
-//! This exercises `TypedAllocator`'s `[u8; N]` allocation behavior, which once
-//! involved a `*const [u8; N]` to `*const u8` pointer cast. Why is this cast
-//! different from all other casts? `crucible-mir`'s custom array representation
-//! means that it needs to insert an array-indexing operation in `*const [T; N]`
-//! to `*const T` casts, even though in this particular case, that operation
-//! results in a mistyped pointer, to which `[u8; N]`s cannot be written. See
-//! https://github.com/GaloisInc/mir-json/pull/148.
-
+/// This exercises `TypedAllocator`'s `[u8; N]` allocation behavior, which once
+/// involved a `*const [u8; N]` to `*const u8` pointer cast. Why is this cast
+/// different from all other casts? `crucible-mir`'s custom array representation
+/// means that it needs to insert an array-indexing operation in `*const [T; N]`
+/// to `*const T` casts, even though in this particular case, that operation
+/// results in a mistyped pointer, to which `[u8; N]`s cannot be written. See
+/// https://github.com/GaloisInc/mir-json/pull/148.
 #[cfg_attr(crux, crux::test)]
-fn test() -> [u8; 4] {
+fn test_push() -> [u8; 4] {
     let mut v: Vec<[u8; 4]> = Vec::new();
     let a: [u8; 4] = [1, 2, 3, 4];
     v.push(a);
     v[0]
 }
 
+/// This exercises `RawVec`'s `[u8; N]` allocation behavior, which had precisely
+/// the same `*const [u8; N]` to `*const u8` casting issue as `TypedAllocator`,
+/// as described above. See https://github.com/GaloisInc/mir-json/pull/150.
+#[cfg_attr(crux, crux::test)]
+fn test_macro() -> [u8; 4] {
+    let mut v: Vec<[u8; 4]> = alloc::vec!([5, 6, 7, 8]);
+    v[0]
+}
+
 fn main() {
-    println!("{:?}", test());
+    println!("{:?}", test_push());
+    println!("{:?}", test_macro());
 }
```

Comment:
Does this actually work with `cabal test`?  I don't think we have any other `conc_eval` tests with multiple `#[crux::test]` functions in the same file.

Reply:
Ah, I didn't even think of that. I was just running `crux-mir` directly on this file.

Local experimentation suggests that this approach is _almost_ compatible with `cabal test`. The test harness seems to have no issues comparing output for multiple tests, but `crux-mir` seems to run a file's tests in alphabetical order by function name: https://github.com/GaloisInc/crucible/blob/5c9211099ec54bd34c179877b7d639d264767374/crux-mir/src/Mir/Language.hs#L290

Seems like that was done in 3f6510c0a0dbb717b56df2f04be9e04e11316ad5 as part of better mimicking `cargo` output. Would you prefer to maintain that behavior and split this up into multiple files, or do you think it'd be acceptable to remove that `List.sort` call?

Reply:
Unsorted output seems like it would be harder to read, so I think it would be better to split the test case into two.